File: CPlusPlus/State/src/SiegeTank.cpp
/*************************************************************************
    > File Name: SiegeTank.cpp
    > Author: ma6174
    > Mail: ma6174@163.com 
    > Created Time: 2017年03月20日 星期一 19时49分53秒
 ************************************************************************/

#include"SiegeTank.h"

using namespace std;

SiegeTank::SiegeTank():pSiegeState(make_shared<SiegeState>()),pTankState(make_shared<TankState>())
{
	pSiegeTankState=pSiegeState;
}

SiegeTank::~SiegeTank()
{
	;
}

void SiegeTank::EnterSiegeMode()
{
	pSiegeTankState=pSiegeState;
}

void SiegeTank::EnterTankMode()
{
	pSiegeTankState=pTankState;
}

void SiegeTank::Move(int x, int y)
{
	pSiegeTankState->Move(x,y);
}

void SiegeTank::Attack()
{
	pSiegeTankState->Attack();
}

File: Python/State/SiegeTank.py
# -*- coding:utf-8 -*-

from SiegeTankState import *

class SiegeTank:

    def __init__(self):
        self.__pSiegeState = SiegeState()
        self.__pTankState = TankState()
        self.__pSiegeTankState = self.__pSiegeState

    def EnterSiegeMode(self):
        self.__pSiegeTankState = self.__pSiegeState

    def EnterTankMode(self):
        self.__pSiegeTankState = self.__pTankState

    def Attack(self):
        self.__pSiegeTankState.Attack()

    def Move(self, x, y):
        self.__pSiegeTankState.Move(x, y)
